// formik_typescript_ui/dist/CheckboxField.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var formik_1 = require("formik");
var CheckboxField = function (_a) {
    var className = _a.className, classNameLabel = _a.classNameLabel, disabled = _a.disabled, id = _a.id, label = _a.label, labelStyle = _a.labelStyle, name = _a.name, style = _a.style;
    var renderCheckbox = function (_a) {
        var _b = _a.field, onChange = _b.onChange, value = _b.value;
        return (react_1.default.createElement(react_1.default.Fragment, null,
            react_1.default.createElement("input", { checked: value, className: className || '', disabled: disabled, id: id || name, name: name, onChange: onChange, style: style, type: "checkbox", value: value }),
            react_1.default.createElement("label", { htmlFor: id || name, className: classNameLabel || '', style: labelStyle }, label)));
    };
    return (react_1.default.createElement(formik_1.Field, { name: name, render: renderCheckbox }));
};
CheckboxField.defaultProps = {
    className: null,
    classNameLabel: null,
    disabled: false,
    id: null,
    labelStyle: {},
    style: {},
};
exports.default = CheckboxField;
