// formik_typescript_ui/dist/DatePickerField.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var formik_1 = require("formik");
var DatePickerField = function (_a) {
    var className = _a.className, disabled = _a.disabled, id = _a.id, maxDate = _a.maxDate, minDate = _a.minDate, name = _a.name, style = _a.style;
    var renderDatepicker = function (_a) {
        var field = _a.field;
        return (react_1.default.createElement("input", __assign({}, field, { className: className || '', disabled: disabled, id: id || name, max: maxDate || '', min: minDate || '', style: style, type: "date" })));
    };
    return (react_1.default.createElement(formik_1.Field, { name: name, render: renderDatepicker }));
};
DatePickerField.defaultProps = {
    className: null,
    disabled: false,
    id: null,
    maxDate: null,
    minDate: null,
    style: {},
};
exports.default = DatePickerField;
