// formik_typescript_ui/dist/TextField.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var react_1 = __importDefault(require("react"));
var formik_1 = require("formik");
var TextField = function (_a) {
    var className = _a.className, disabled = _a.disabled, forwardRef = _a.forwardRef, id = _a.id, name = _a.name, max = _a.max, maxLength = _a.maxLength, min = _a.min, minLength = _a.minLength, onChange = _a.onChange, onInput = _a.onInput, onKeyDown = _a.onKeyDown, pattern = _a.pattern, placeholder = _a.placeholder, style = _a.style, type = _a.type;
    var renderInput = function (_a) {
        var field = _a.field;
        return (react_1.default.createElement("input", __assign({}, field, { className: className || '', disabled: disabled, id: id || name, max: max, maxLength: maxLength, min: min, minLength: minLength, onChange: onChange || field.onChange, onInput: onInput, onKeyDown: onKeyDown, pattern: pattern, placeholder: placeholder || '', ref: forwardRef, style: style, type: type })));
    };
    return (react_1.default.createElement(formik_1.Field, { type: type, name: name, render: renderInput }));
};
TextField.defaultProps = {
    className: null,
    disabled: false,
    forwardRef: undefined,
    id: null,
    max: undefined,
    maxLength: undefined,
    min: undefined,
    minLength: undefined,
    onChange: undefined,
    onInput: undefined,
    onKeyDown: undefined,
    pattern: undefined,
    placeholder: null,
    style: {},
    type: 'text',
};
exports.default = TextField;
